src/lib.rs

```diff
@@ -625,7 +625,9 @@ impl<'a, R: ?Sized> Rng for &'a mut R where R: Rng {
     }
 }
 
-#[cfg(feature="std")]
+#[cfg(all(feature="alloc", not(feature="std")))]
+use alloc::boxed::Box;
```

Comment:
I'd prefer if you put the `use` statement above, near `#[cfg(feature="std")] use std::rc::Rc;`.